lizzy_client/cli.py

```diff
@@ -258,26 +290,50 @@ def traffic(stack_name: str, stack_version: str, percentage: int, remote: str):
 
 
 @main.command()
-@click.argument('stack_name')
-@click.argument('stack_version')
+@click.argument('stack_ref', nargs=-1)
+@region_option
+@click.option('--dry-run', is_flag=True, help='No-op mode: show what would be deleted')
+@click.option('-f', '--force', is_flag=True, help='Allow deleting multiple stacks')
 @remote_option
-def delete(stack_name: str, stack_version: str, remote: str):
-    '''Delete a single stack'''
+def delete(stack_ref: List[str],
+           region: str, dry_run: bool, force: bool, remote: str):
+    """Delete Cloud Formation stacks"""
     config = Configuration()
 
     access_token = fetch_token(config.token_url, config.scopes, config.credentials_dir)
 
     lizzy_url = remote or config.lizzy_url
     lizzy = Lizzy(lizzy_url, access_token)
 
-    with Action('Requesting stack deletion..'):
-        stack_id = '{stack_name}-{stack_version}'.format_map(locals())
-        try:
-            lizzy.delete(stack_id)
-        except requests.ConnectionError as e:
-            connection_error(e)
-        except requests.HTTPError as e:
-            agent_error(e)
+    stack_refs = get_stack_refs(stack_ref)
+
+    all_with_version = all(stack.version is not None
+                           for stack in stack_refs)
+
+    # TODO unit test this
+    #  this is misleading but it's the current behaviour of senza
+    # TODO Lizzy list (stack_refs) to see if it actually matches more than one stack
+    # to match senza behaviour
+    if (not all_with_version and not dry_run and not force):
+        fatal_error('Error: {} matching stacks found. '.format(len(stack_refs)) +
+                    'Please use the "--force" flag if you really want to delete multiple stacks.')
+
+    # TODO pass force
+
+    for stack in stack_refs:
```

Comment:
You can only delete a single resource at a time with a REST API. Remember the Lizzy Agent receives a `DELETE` in `api/{stack_id}` so it can't receive a list.